atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraVerifier.java

```diff
@@ -265,106 +285,176 @@ private static boolean attemptToCreateKeyspaceOnHost(InetSocketAddress host, Cas
+                cassandraKeyspaceVerifierConfig.ignoreNodeTopologyChecks());
+        sanityCheckReplicationFactor(
+                ks,
+                cassandraKeyspaceVerifierConfig.replicationFactor(),
+                dcs,
+                cassandraKeyspaceVerifierConfig.ignoreDatacenterConfigurationChecks());
     }
 
-    static void sanityCheckReplicationFactor(KsDef ks, CassandraKeyValueServiceConfig config, Set<String> dcs) {
+    static void sanityCheckReplicationFactor(
+            KsDef ks, int replicationFactor, Set<String> dcs, boolean ignoreDatacentreConfigurationChecks) {
         Set<String> scopedDownDcs = checkRfsSpecifiedAndScopeDownDcs(dcs, ks.getStrategy_options());
-        checkRfsMatchConfig(ks, config, scopedDownDcs, ks.getStrategy_options());
+        checkRfsMatchConfig(
+                ks, replicationFactor, scopedDownDcs, ks.getStrategy_options(), ignoreDatacentreConfigurationChecks);
     }
 
     private static Set<String> checkRfsSpecifiedAndScopeDownDcs(Set<String> dcs, Map<String, String> strategyOptions) {
         return Sets.intersection(dcs, strategyOptions.keySet());
     }
 
     private static void checkRfsMatchConfig(
-            KsDef ks, CassandraKeyValueServiceConfig config, Set<String> dcs, Map<String, String> strategyOptions) {
+            KsDef ks,
+            int replicationFactor,
+            Set<String> dcs,
+            Map<String, String> strategyOptions,
+            boolean ignoreDatacenterConfigurationChecks) {
         for (String datacenter : dcs) {
-            if (Integer.parseInt(strategyOptions.get(datacenter)) != config.replicationFactor()) {
+            if (Integer.parseInt(strategyOptions.get(datacenter)) != replicationFactor) {
                 logErrorOrThrow(
                         "Your current Cassandra keyspace (" + ks.getName()
                                 + ") has a replication factor not matching your Atlas Cassandra configuration."
                                 + " Change them to match, but be mindful of what steps you'll need to"
                                 + " take to correctly repair or cleanup existing data in your cluster.",
-                        config.ignoreDatacenterConfigurationChecks());
+                        ignoreDatacenterConfigurationChecks);
             }
         }
     }
+
+    @Value.Immutable
+    interface CassandraKeyspaceVerifierConfig {
+        String keyspace();
+
+        CassandraKeyValueServiceConfig clientConfig();
+
+        CassandraServersConfig servers();
+
+        int replicationFactor();
+
+        boolean ignoreNodeTopologyChecks();
+
+        boolean ignoreDatacenterConfigurationChecks();
+
+        int schemaMutationTimeoutMillis();
+
+        static CassandraKeyspaceVerifierConfig of(CassandraKeyValueServiceConfig config) {
+            return builder()
+                    .keyspace(config.getKeyspaceOrThrow())
+                    .clientConfig(config)
```

Comment:
I see that we use it in `CassandraClientFactory` calls; as mentioned, I think it's worth making the values lazy here at the very least.

Reply:
Yeah this was my bad, I forgot to mention this (since I split up the changes) - there's another PR to change for clientConfig just to be a scoped down version of the config, but to keep this one small I decided to keep the full config for just `CassandraClientFactory`, and then in the next PR I only need to change 2 lines in the builder to use a `CassandraClientConfig`.